### PR TITLE
NFC: silence a few warnings

### PR DESCRIPTION
**Pull request by davezarzycki**

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please smoke test and merge

File: lib/AST/GenericSignatureBuilder.cpp [hunk 1]
```diff
@@ -587,11 +587,13 @@ namespace {
       return result;
     }
 
+    LLVM_ATTRIBUTE_UNUSED // If unused, silence -Wunused-function
```
Comment (slavapestov) [inline lib/AST/GenericSignatureBuilder.cpp]: This whole class definition is dead actually. See https://github.com/apple/swift/pull/27535. Do you want to incorporate my change into your PR?

Comment (davezarzycki) [reply lib/AST/GenericSignatureBuilder.cpp]: Are you tired of begging CI to try again? :-P

But sure, ya, these two PRs will conflict. I'll update this PR.

**Comment by davezarzycki** [comment]
@swift-ci please smoke test

**Comment by davezarzycki** [comment]
@swift-ci please clean test linux

**Comment by davezarzycki** [comment]
@swift-ci please smoke test linux

**Comment by davezarzycki** [comment]
@swift-ci please smoke test linux

**Comment by davezarzycki** [comment]
What's going on with the unrelated Foundation test failures?

@swift-ci please smoke test

**Comment by theblixguy** [comment]
Weird - the entire URLSession test suite was [disabled](https://github.com/apple/swift-corelibs-foundation/pull/2533) yesterday so this failures shouldn't exist.

**Pull request merged**